rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -229,7 +229,7 @@ def __iter__(self):
         return self
 
     def __len__(self):
-        return self.iterator_length
+        return self.iterator_length // self.batch_size
```

Comment:
Is this a fix?

Reply:
Yes. This is used in diffusion training tqdm bar. Without this, it gives the length as the total number of images per shard

Reply:
But will this have an impact on other trainings? What should len return?

Reply:
This is the correct value for the length of the iterator and this is what we should return. Currently, in classification trainings, we divide the length by batch size in training pipeline and print it. I can modify the code there